src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs

```diff
@@ -4824,6 +4824,188 @@ public static D Dynamic_b_not_null_a_nullable(int? c)
 }
 ";
             var compilation = CreateCompilation(source).VerifyDiagnostics();
+            compilation.GetEmitDiagnostics();
+        }
+
+        [Fact, WorkItem(60059, "https://github.com/dotnet/roslyn/issues/60059")]
+        public void TestNullCoalesce_NullableIntAndDynamic()
+        {
+            var source = @"
+int? i = 42;
+dynamic d = new object();
+System.Console.Write(i ?? d);
+";
+            var compilation = CreateCompilation(source, references: new[] { CSharpRef });
+            compilation.VerifyDiagnostics();
+            compilation.GetEmitDiagnostics();
```

Comment:
Similarly: is this needed when we have the VerifyUsedAssemblies leg? And if we do need it, should we verify the set of diagnostics here? Same nit for the below calls to 'GetEmitDiagnostics'.